sdl_android_lib/src/com/smartdevicelink/transport/SdlPsm.java

```diff
@@ -153,7 +155,11 @@ private int transitionOnInput(byte rawByte, int state){
 					if(dataLength==0){
 						return FINISHED_STATE; //We are done if we don't have any payload
 					}
-					payload = new byte[dataLength];
+					if(dataLength <= V1_V2_MTU_SIZE){ // size from protocol/WiProProtocol.java
```

Comment:
This should actually be MTU-HEADER_SIZE. Introduce statics into WiProProtocol for standard Header sizes. V1_HEADER_SIZE = 8, V2_V4_HEADER_SIZE = 12